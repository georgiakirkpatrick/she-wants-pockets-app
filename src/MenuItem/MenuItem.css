.empty {
    height: 0;
    width: 0;
}

.MenuItem__primary, .MenuItem__secondary, .MenuItem__secondary-menu-header {
    font-size: 18px;
    height: 50px;
    width: 100%;
    padding: 0 30px;
    display: flex;
    align-items: center;
    border-radius: var(--border-radius);
    transition: background var(--speed);
    padding: 0.5rem;
}

.MenuItem__primary, .MenuItem__secondary {
    justify-content: flex-start;
}

.MenuItem__secondary-menu-header {
    font-weight: bold;
}


.MenuItem__icon-left {
    margin-right: auto;
}

.MenuItem__icon-right {
    margin-left: auto;
}

@media screen and (min-width: 768px) {   
    .MenuItem__primary {
        height: 45px;
        padding: 0;
    }
}