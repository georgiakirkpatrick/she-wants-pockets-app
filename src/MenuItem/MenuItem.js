import React from 'react'
import { BrowserRouter as Router, Link } from 'react-router-dom'
import './MenuItem.css'

const MenuItem = props => {
    const leftIcon = props.leftIcon 
        ? <span className='MenuItem__icon-left'>{props.leftIcon}</span>
        : <div className='empty' />

    const rightIcon = props.rightIcon 
        ? <span className='MenuItem__icon-right'>{props.rightIcon}</span>
        : <div className='empty' />
    
    return (
    <Router>
        <Link
            className={`MenuItem__${props.itemType}`}
            to={props.to}
            onClick={() => props.goToMenu && props.setActiveMenu(props.goToMenu)}
        >
            {leftIcon}
            <span>{props.children}</span>
            {rightIcon}
        </Link>
    </Router>
    )
}

MenuItem.defaultProps = {
    leftIcon: '',
    rightIcon: '',
    itemType: 'primary',
    to: '/',
    goToMenu: 'main',
    setActiveMenu: () => {}
}

export default MenuItem