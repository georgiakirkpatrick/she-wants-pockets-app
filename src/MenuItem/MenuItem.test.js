import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import MenuItem from './MenuItem'
import { BrowserRouter as Router } from 'react-router-dom'

describe ('MenuItem', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Router><MenuItem /></Router>, div
        )
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Router><MenuItem /></Router>)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})