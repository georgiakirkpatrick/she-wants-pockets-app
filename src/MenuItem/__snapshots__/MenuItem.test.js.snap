// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem renders the UI as expected 1`] = `
<a
  className="MenuItem__primary"
  href="/"
  onClick={[Function]}
>
  <div
    className="empty"
  />
  <span />
  <div
    className="empty"
  />
</a>
`;
